airflow/models/dag.py

```diff
@@ -1378,6 +1379,37 @@ def normalized_schedule_interval(self) -> ScheduleInterval:
             _schedule_interval = self.schedule_interval
         return _schedule_interval
 
+    @staticmethod
+    @internal_api_call
+    @provide_session
+    def fetch_callback(
+        dag: DAG,
+        dagrun: DagRun | DagRunPydantic,
+        success: bool = True,
+        reason: str | None = None,
+        session: Session = NEW_SESSION,
+    ):
```

Comment:
```suggestion
        *,
        session: Session = NEW_SESSION,
    ) -> None | tuple[TaskStateChangeCallback | list[TaskStateChangeCallback], Context]:
```

Reply:
If I add annotations then mypy will analyse the code and mypy-core will complain about:

```
airflow/models/dag.py:1411: error: TypedDict "Context" has no key "reason"
[typeddict-unknown-key]
                context["reason"] = reason
```

I spent a lot of time trying to resolve it with no chance. If you have any solution I am definitely interested. The only solution I could find to bypass this problem is to not add annotations. This might feel like I am hiding an issue, but this is how it is today, I dont introduce something new, I just keep it as is

Reply:
Actually, thanks to [your comment](https://github.com/apache/airflow/pull/28900#discussion_r1260380579) I found out how. Done